src/structures_node.jl

```diff
@@ -20,26 +20,27 @@ abstract type Availability <: NetworkNode end
 - **`cap::TimeProfile`** is the installed capacity.\n
 - **`opex_var::TimeProfile`** is the variational operational costs per energy unit produced.\n
 - **`opex_fixed::TimeProfile`** is the fixed operational costs.\n
-- **`output::Dict{Resource, Real}`** are the generated `Resource`s with conversion value `Real`.\n
-- **`data::Array{Data}`** is the additional data (e.g. for investments).
+- **`output::Dict{<:Resource, <:Real}`** are the generated `Resource`s with conversion value `Real`.\n
+- **`data::Vector{Data}`** is the additional data (e.g. for investments). The field \
+`data` is conditional through usage of a constructor.
 
 """
 struct RefSource <: Source
     id
     cap::TimeProfile
     opex_var::TimeProfile
     opex_fixed::TimeProfile
-    output::Dict{Resource, Real}
-    data::Array{Data}
+    output::Dict{<:Resource, <:Real}
+    data::Vector{Data}
 end
 function RefSource(
-    id,
-    cap::TimeProfile,
-    opex_var::TimeProfile,
-    opex_fixed::TimeProfile,
-    output::Dict{<:Resource,<:Real},
+        id,
+        cap::TimeProfile,
+        opex_var::TimeProfile,
+        opex_fixed::TimeProfile,
+        output::Dict{<:Resource,<:Real},
```

Comment:
Did the formatter add this indent?

Reply:
No, as I do not use formatter for EMB. That I added manually, but, honestly, I do not know, why. In practice, where not entirely consistent in white spacing around function arguments. But I do not want to add the formatter, as there are quite a few `@constraint`s which I think the formatter makes difficult to read.